src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala

```diff
@@ -0,0 +1,115 @@
+  "PodsController" should {
+    "support pods" in {
+      val controller = Fixture().controller()
+      Head(Uri./) ~> controller.route ~> check {
+        response.status should be(StatusCodes.OK)
+        responseAs[String] shouldBe empty
+      }
+    }
+
+    {
+      val controller = Fixture(authenticated = false).controller()
+      behave like unauthenticatedRoute(forRoute = controller.route, withRequest = Head(Uri./))
+      behave like unauthenticatedRoute(forRoute = controller.route, withRequest = Post(Uri./))
+    }
+
+    {
+      val controller = Fixture(authorized = false).controller()
+      val podSpecJson = """
+                          | { "id": "/mypod", "networks": [ { "mode": "host" } ], "containers": [
+                          |   { "name": "webapp",
+                          |     "resources": { "cpus": 0.03, "mem": 64 },
+                          |     "image": { "kind": "DOCKER", "id": "busybox" },
+                          |     "exec": { "command": { "shell": "sleep 1" } } } ] }
+                        """.stripMargin
+      val entity = HttpEntity(podSpecJson).withContentType(ContentTypes.`application/json`)
+      val request = Post(Uri./.withQuery(Query("force" -> "false")))
+        .withEntity(entity)
+        .withHeaders(`Remote-Address`(RemoteAddress(InetAddress.getByName("192.168.3.12"))))
+      behave like unauthorizedRoute(forRoute = controller.route, withRequest = request)
+    }
+
+    "be able to create a simple single-container pod from docker image w/ shell command" in {
+      val f = Fixture(configArgs = Seq("--default_network_name", "blah")) // should not be injected into host network spec
+      val controller = f.controller()
+
+      val deploymentPlan = DeploymentPlan.empty
+      f.podManager.create(any, eq(false)).returns(Future.successful(deploymentPlan))
+
+      val podSpecJson = """
+                          | { "id": "/mypod", "networks": [ { "mode": "host" } ], "containers": [
+                          |   { "name": "webapp",
+                          |     "resources": { "cpus": 0.03, "mem": 64 },
+                          |     "image": { "kind": "DOCKER", "id": "busybox" },
+                          |     "exec": { "command": { "shell": "sleep 1" } } } ] }
+                        """.stripMargin
+      val entity = HttpEntity(podSpecJson).withContentType(ContentTypes.`application/json`)
+      val request = Post(Uri./.withQuery(Query("force" -> "false")))
+        .withEntity(entity)
+        .withHeaders(`Remote-Address`(RemoteAddress(InetAddress.getByName("192.168.3.12"))))
+      request ~> controller.route ~> check {
+        response.status should be(StatusCodes.Created)
+        response.header[Headers.`Marathon-Deployment-Id`].value.value() should be(deploymentPlan.id)
+        response.header[Location].value.value() should be("/mypod")
+
+        (Json.parse(responseAs[String]) \ "networks" \ 0 \ "mode") shouldBe JsDefined(JsString(raml.NetworkMode.Host.value))
```

Comment:
Nit: you could probably parse the response only once here.

Reply:
There is a follow up which makes this much nicer.